src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/memberdays/MemberDaysReport.java

```diff
@@ -0,0 +1,312 @@
+import uk.gov.hmcts.ecm.common.model.ccd.CaseData;
+import uk.gov.hmcts.ecm.common.model.ccd.SubmitEvent;
+import uk.gov.hmcts.ecm.common.model.ccd.items.DateListedTypeItem;
+import uk.gov.hmcts.ecm.common.model.ccd.items.HearingTypeItem;
+import uk.gov.hmcts.ecm.common.model.listing.ListingDetails;
+
+import java.time.Duration;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.groupingBy;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.HEARING_STATUS_HEARD;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.MEMBER_DAYS_REPORT;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.SINGLE_HEARING_DATE_TYPE;
+
+@Service
+@Slf4j
+public class MemberDaysReport {
+    private static final int MINUTES = 60;
+    private List<SubmitEvent> submitEvents;
+    private ListingDetails listingDetails;
+    private static final String FULL_PANEL = "Full Panel";
+
+    public MemberDaysReportData runReport(ListingDetails listings, List<SubmitEvent> submitEventList) {
+        submitEvents = submitEventList;
+        listingDetails = listings;
+
+        var memberDaysReportData = initiateReport(listingDetails.getCaseTypeId());
+        addReportDetails(memberDaysReportData);
+        addReportSummary(memberDaysReportData);
+        addReportSummaryHeader(memberDaysReportData);
+
+        return memberDaysReportData;
+    }
+
+    private MemberDaysReportData initiateReport(String caseTypeId) {
+        var office = UtilHelper.getListingCaseTypeId(caseTypeId);
+        var reportData = new MemberDaysReportData();
+        reportData.setOffice(office);
+        reportData.setDurationDescription(getDurationText(reportData));
+        reportData.setReportType(MEMBER_DAYS_REPORT);
+        return reportData;
+    }
+
+    private String getDurationText(MemberDaysReportData reportData) {
+        var description = "";
+        var currentCaseData = listingDetails.getCaseData();
+        if (currentCaseData.getHearingDateType().equals("Single")) {
+            description = "On " + currentCaseData.getListingDate();
+        } else if (currentCaseData.getHearingDateType().equals("Range")) {
+            description = "Between " + currentCaseData.getListingDateFrom()
+                    + " and " + currentCaseData.getListingDateTo();
+        }
+        reportData.setHearingDateType(currentCaseData.getHearingDateType());
+        return description;
+    }
+
+    private void addReportDetails(MemberDaysReportData reportData) {
+        List<MemberDaysReportDetail> interimReportDetails = new ArrayList<>();
+        for (var submitEvent : submitEvents) {
+
+            if (!isValidCase(submitEvent)) {
+                continue;
+            }
+
+            var caseData = submitEvent.getCaseData();
+            log.info("Adding case {} to Member Days report", caseData.getEthosCaseReference());
```

Comment:
How many cases could you see on a report? Are we going to get logs full of large numbers of these messages?

Reply:
It is potentially(in theory) high as this section lists every single matching hearing date, that could be multiple even in one case. Hence, I have taken out the log message.